wpilibcExamples/src/main/cpp/examples/TrajectoryCommand/cpp/subsystems/DriveSubsystem.cpp

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include "subsystems/DriveSubsystem.h"
+
+#include <frc/geometry/Rotation2d.h>
+#include <frc/kinematics/ChassisSpeeds.h>
+#include <frc/kinematics/DifferentialDriveWheelSpeeds.h>
+#include <frc2/command/InstantCommand.h>
+#include <frc2/command/SequentialCommandGroup.h>
+#include <frc2/command/TrajectoryCommand.h>
+#include <units/voltage.h>
+
+using namespace DriveConstants;
+using namespace AutoConstants;
+
+DriveSubsystem::DriveSubsystem()
+    : m_left1{kLeftMotor1Port},
+      m_left2{kLeftMotor2Port},
+      m_right1{kRightMotor1Port},
+      m_right2{kRightMotor2Port},
+      m_leftEncoder{kLeftEncoderPorts[0], kLeftEncoderPorts[1]},
+      m_rightEncoder{kRightEncoderPorts[0], kRightEncoderPorts[1]},
+      m_ramseteController{kRamseteB, kRamseteZeta},
+      m_leftController{kPDriveVel, 0, 0},
+      m_rightController{kPDriveVel, 0, 0},
+      m_feedforward{kS, kV, kA},
+      m_odometry{m_gyro.GetRotation2d()} {
+  // Set the distance per pulse for the encoders
+  m_leftEncoder.SetDistancePerPulse(kEncoderDistancePerPulse);
+  m_rightEncoder.SetDistancePerPulse(kEncoderDistancePerPulse);
+
+  ResetEncoders();
+}
+
+void DriveSubsystem::Periodic() {
+  // Implementation of subsystem periodic method goes here.
+  m_odometry.Update(m_gyro.GetRotation2d(),
+                    units::meter_t(m_leftEncoder.GetDistance()),
+                    units::meter_t(m_rightEncoder.GetDistance()));
+}
+
+void DriveSubsystem::ArcadeDrive(double fwd, double rot) {
+  m_drive.ArcadeDrive(fwd, rot);
+}
+
+void DriveSubsystem::FollowState(const frc::Trajectory::State& targetState) {
+  // Calculate via Ramsete the speed vector needed to reach the target state
+  frc::ChassisSpeeds speedVector =
+      m_ramseteController.Calculate(m_odometry.GetPose(), targetState);
+  // Convert the vector to the separate velocities of each side of the
+  // drivetrain
+  frc::DifferentialDriveWheelSpeeds targetSpeeds =
+      kDriveKinematics.ToWheelSpeeds(speedVector);
+
+  // PID and Feedforward control
+  // This can be replaced by calls to smart motor controller closed-loop control
+  // functionality. For example:
+  // mySmartMotorController.setSetpoint(leftMetersPerSecond,
+  // ControlType.Velocity);
+
+  auto currentLeftVelocity = m_leftEncoder.GetRate();
+  auto currentRightVelocity = m_rightEncoder.GetRate();
+
+  auto rightFeedforward = m_feedforward.Calculate(
+      currentLeftVelocity, targetSpeeds.left, AutoConstants::kTimestep);
```

Comment:
This should be `leftFeedforward`. Also, the measurement shouldn't be passed to the feedforward. Only setpoints should be.